Project-3

Gizmofire opened the pull request.

Commit:
Added more to readme

ThomasHakwins committed:
possible functions to use

Just came up with a bunch of functions for finding loot

Gizmofire committed:
new class and funcs

ThomasHakwins committed:
added some items to craft

Gizmofire committed:
update

Gizmofire committed:
main update

Gizmofire committed:
Merge remote-tracking branch 'origin/backpack-items'

Gizmofire committed:
figuered out methods

ThomasHakwins committed:
added intro and a few other things

Gizmofire committed:
update to backpack

Commit:
Merge remote-tracking branch 'origin/backpack-items'

Commit:
team update

ThomasHakwins committed:
added enemies and fighting

ThomasHakwins committed:
added fighting and enemies

Gizmofire committed:
cleaned up backpack upgrading

Gizmofire committed:
FINISED CRAFTING portion

Gizmofire committed:
proof it works

ThomasHakwins committed:
done with gameplay functions and commenting?

ThomasHakwins committed:
added extras stuff

ThomasHakwins committed:
new fighting function

Commit:
backpack tests

Commit:
Merge branch 'main' of https://github.com/ETE4990-S25/project-1-gordon-and-thomas-in-the-mornin

Commit:
done with integration with thomas

Commit:
some more QOL fixes

Commit:
added descrition and readme

Commit:
readme small update

ThomasHakwins committed:
updated readme

ThomasHakwins committed:
reupdated readme

Commit:
added proof of game

Commit:
Merge branch 'main' of https://github.com/ETE4990-S25/project-1-gordon-and-thomas-in-the-mornin

Commit:
clean up of read me

ThomasHakwins committed:
added player class option

ThomasHakwins committed:
Merge branch 'main' of https://github.com/ETE4990-S25/project-1-gordon-and-thomas-in-the-mornin

Gizmofire committed:
added saving for gender

philo0o0 commented on game.py:
```diff
@@ -0,0 +1,263 @@
+    return angle
+
+# Basic function for moving around the map and finding enemies
+def move_Direction(bp):
```
there is nothing to exit with from here I would add a condition to exit

philo0o0 commented on game.py:
```diff
@@ -0,0 +1,263 @@
+    low_Loot_number_generator = random.randint(1, 5)
+    amount_of_items = random.randint(1, 3)
+    if low_Loot_number_generator == 1:
+        (lambda items: print(f"You found {items} {'stick' if items == 1 else 'sticks'}"))(amount_of_items)
```
so you have good ideas on line 99  but a clever way to make it easier on yourself would be to lose the lambda function since it is not really needed here and it would make your code a little cleaner without it

Gizmofire replied:
As per suggestion. 

```suggestion
        if (amount_of_items == 1):
            print(f"You found a stick")
        else:
            print(f"You found {amount_of_items} sticks")
```

philo0o0 commented on game.py:
```diff
@@ -0,0 +1,263 @@
+            else:
+                print("Invalid input. Please choose 'male' or 'female'.")
+
+        def save_gender_to_file(self, filename="gender.json"):
```
line 261 appears to be indented inside of choose gender which will make it not accessible outside of the character class

philo0o0 commented on game.py:
```diff
@@ -0,0 +1,263 @@
+            
+        print(f"Wins to beat: {wins}/{wins_needed}")
+        print(f"Reminder: You need {wins_needed} wins to defeat the {enemy.name}")
+    
```
here I would add some factor in the game where if they lose consecutively something happens or else the game just keeps letting them retry until they win which will make the win not feel good, the loses make the win all the better

philo0o0 commented on game.py:
```diff
@@ -0,0 +1,263 @@
+        
+        if random.randint(1, 6) == 1:
+            print("You were attacked by a wild animal")
+            fighting(bp)
```
so good Idea on running into something and taking action but you also don't want to fight endlessly so I would also add a break option in order to continue with the game ahead

philo0o0 commented on game.py:
```diff
@@ -0,0 +1,263 @@
+    player = Character()  # Create a character instance and choose gender
+    play = True
+    while play:
+        print("\n--- Welcome to the Adventure Game ---")
```
on all of these you forgot to close the parenthesis

philo0o0 commented on game.py:
```diff
@@ -0,0 +1,263 @@
+        print("4) Craft Items")
+        print("5) Exit and Save")
+        print("10) HARD RESET BACKPACK")
+        
```
lines 115 -120 would be better suited in a dictionary where you can easily expand the game in the future and then you can add what the menu option does then add the action where it would be easily accessible if you ever want to add more options or edit and existing options you won't have to go into the main code and search for what you need to add or edit

philo0o0 commented on game.py:
```diff
@@ -0,0 +1,263 @@
+    else:
+        angle = math.degrees(math.atan(position[1] / position[0]))
+    return angle
+
```
I really like your moving settings I would just maybe add a break after so they are not moving endlessly or maybe if they run into something they can get the option to take an action there.

Gizmofire committed:
Addressed https://github.com/ETE4990-S25/project-1-gordon-and-thomas-in-the-mornin/pull/3#discussion_r2017829394

JuanEstrada1 commented on main.py:
is this where the main game is supposed to be played on? There is another file called "game.py" so leave a comment where the player should start the game at and whether the files will be imported or change the file names. I see that the explanation is in the "readme.md" file, so this is more of a recommendation.

JuanEstrada1 commented on project stuff.ipynb:
Recommend storing all work-in-progress code and junk into a separate folder so that I can be organized and out of sight as to not cause any confusion for people who will look through this code.

philo0o0 commented on items_and_backpack/backpack.py:
```diff
@@ -0,0 +1,130 @@
+            if backpack_item['name'] == item:
+                backpack_item['count'] += count
+                item_added = True
+                self.save_backpack
```
here on line 27 you are intending to call the function but you are missing the parenthesis here which means it is not really calling the function you, should have have it the same just with parenthesis so it should look something like this self.save_backpack()

philo0o0 commented on items_and_backpack/backpack.py:
```diff
@@ -0,0 +1,130 @@
+                    return item['count']
+        except:
+            return 0
+
```
the except function here takes up all the exceptions, it would be beneficial to catch each exception for 2 main reasons, one is that it would make your code cleaner and two it would make debugging much easier since it won't cover up anything.

philo0o0 commented on items_and_backpack/backpack.py:
```diff
@@ -0,0 +1,130 @@
+    
+    # print(bp.toString())
+    # bp.reset_backpack()
+
```
this code is commented out and it is a little confusing if you did this on purpose or if this was an accident, if it was on purpose I would add a comment at the end to help define what exactly it is, if it is not supposed to be commented out then my comment is just to put it in the code instead of comment it.

philo0o0 commented on items_and_backpack/crafting.py:
```diff
@@ -0,0 +1,111 @@
+    for i, item in enumerate(pos_items, start=1):
+        print(f"{i}) {item}")
+    print("Enter '0' to go back")
+    
```
for your crafting lines here I would validate the user input as numeric and also have a range set, I would also add an except value error line too so the game doesn't just crash if it runs into a value error or syntax error.

philo0o0 commented on project stuff.ipynb:
on line 86 and the lines above that generate the loot look to be properly coded but I would code it in a dictionary for 2 main reasons, 1 it would make your code much cleaner, and 2 it would make the game much easier to add and remove loot if there is an update for example you would not have to run through your code again to find this specific part, you would just go straight to the dictionary.

philo0o0 commented on project stuff.ipynb:
on line 145 where the battles take place, the wins and losses are accumulated and you take into account fighting harder enemies but I would also like to see you take into account the classes that the player chooses, you can maybe make some classes have an advantage against specific enemies the same way you have an advantage for harder enemies

philo0o0 commented on items_and_backpack/backpack_test.py:
```diff
@@ -0,0 +1,72 @@
+    #  method to run after each to test to reset the backpack state to none
+    def tearDown(self):
+        self.backpack.reset_backpack()
+
```
along with all the backpack tests that you have I would also probably add a test specifically for capacity, you have a test that kind of plays a role with capacity like the not having enough space test but I would maybe add a specific test that tests for the capacity limits of the back pack.

philo0o0 commented on items_and_backpack/backpack_test.py:
```diff
@@ -0,0 +1,72 @@
+    def setUp(self):
+        self.backpack = Backpack("test_backpack.json")
+        self.backpack.reset_backpack()
+
```
along with the testing of the stick on line 13 here I would also maybe list the possible items that could be in the inventory, and if you have the items all in one place like a dictionary this would be an easy place to implement improvements in the code using a dictionary

Gizmofire committed:
Addressing https://github.com/ETE4990-S25/project-1-gordon-and-thomas-in-the-mornin/pull/3#discussion_r2031259554 removed old test cases

Gizmofire committed:
Removed old obsolete file. Used to be the documentation for testing but resulted to just using commit history to show R&D process.

Addresses: https://github.com/ETE4990-S25/project-1-gordon-and-thomas-in-the-mornin/pull/3#discussion_r2029099405

https://github.com/ETE4990-S25/project-1-gordon-and-thomas-in-the-mornin/pull/3#discussion_r2031491192

https://github.com/ETE4990-S25/project-1-gordon-and-thomas-in-the-mornin/pull/3#discussion_r2031522584

JuanEstrada1 commented on items_and_backpack/crafting.py:
```diff
@@ -0,0 +1,111 @@
+    Returns:
+        list: A list of craftable item names.
+    """
+    possible_items = []
```
Lines 44 - 62: See if there is some possible simplification and refactoring for the lines of code.

Ex:
possible_items = []
    for item in crafting_list():
        if all(backpack.get_item_count(n['name']) >= n['count'] for n in item['items_needed']):
            possible_items.append(item['name'])
    return possible_items

JuanEstrada1 commented on items_and_backpack/crafting.py:
```diff
@@ -0,0 +1,111 @@
+        bp (Backpack): The player's backpack.
+        item_name (str): The name of the item to craft.
+    """
+    for item in iM.items:
```
Bit of redundancy for the code of "items_needed." See if you can simplify the code so that it's compressed and easier to manage and read.

Ex:

items_needed = item.get_item_items_needed()
for needed in items_needed:
    if bp.get_item_count(needed) < 1:  # Assuming 1 is the count needed
        print(f"You do not have enough {needed} to craft {item_name}")
        return
    bp.remove_item(needed, 1)

Gizmofire committed:
Added peer review link for gordon

JuanEstrada1 commented on items_and_backpack/items.py:
```diff
@@ -0,0 +1,61 @@
+import json
+
```
Lines 3 - 23: Lines could be changed so that it could be simplified and cleaner to see since there are multiple functions that could be changed.

Ex:

class Item:
    
    def __init__(self, name: str, description: str, items_needed: list, rarity: str):
        self.name = name
        self.description = description
        self.items_needed = items_needed
        self.rarity = rarity

    def __repr__(self):
        return f"Item(name={self.name}, rarity={self.rarity})"

JuanEstrada1 commented on items_and_backpack/items.py:
```diff
@@ -0,0 +1,61 @@
+        return self.rarity
+    
+      
+    
```
Lines 27 - 47: Code seems fine, though you could see if you can simplify the code as to make it readable

Ex:

    class ItemManager:
        def __init__(self, file_path):
            self.file_path = file_path
            with open(file_path, 'r') as file:
                items_data = json.load(file)
                self.items = [Item(
                    name=item['name'],
                    description=item['description'],
                    items_needed=item['itemsNeeded'],
                    rarity=item['rarity']
                ) for item in items_data]

    def get_item(self, name):
        for item in self.items:
            if item.name == name:
                return item
        raise ValueError(f"Item with name '{name}' not found")

JuanEstrada1 commented on items_and_backpack/items.py:
```diff
@@ -0,0 +1,61 @@
+    
+
+
+
```
Lines 54 - 59: Could use this in a separate .py file as to not disrupt the files that can affect the main game.

JuanEstrada1 commented on main.py:
Differentiate between the main files where the game will be played on. If this is not needed then this file could be transferred to a separate folder where work-in-progress files can be stored in.

JuanEstrada1 commented on proof.ipynb:
Could be transferred to another folder where unused and tested files can be stored at.

JuanEstrada1 commented on crafting list.json:
Different dictionaries could be separated into their own sets so that items can be distinguished, such as different dictionaries for rarity of items.

Tangalang22 commented on game.py:
```diff
@@ -0,0 +1,271 @@
+import random
+import math
+from items_and_backpack.crafting import crafting_menu
```
I have no idea if this is just a me issue but I had an error with loading stuff from items_anad_backpack.crafting.

Tangalang22 commented on game.py:
```diff
@@ -0,0 +1,271 @@
+import random
+import math
+from items_and_backpack.crafting import crafting_menu
+from items_and_backpack.backpack import Backpack
```
Backpack doesn't appear to be called at all here. If that's the case, this line can be removed.

Tangalang22 commented on game.py:
```diff
@@ -0,0 +1,271 @@
+
+        bp.add_item("rock", amount_of_items)
+    elif low_Loot_number_generator == 3:
+        print("You found nothing")
```
Is the elif for == 3 needed if the printed statement is the same as the statement for else?

Tangalang22 commented on readme.md:
```diff
@@ -1,6 +1,65 @@
-# Project 1 - Oooga Booga Farming Simulator
+# Project 1 - Oooga Booga Fighting Simulator
```
Love the name and very nice looking readme.

Tangalang22 commented on game.py:
```diff
@@ -0,0 +1,271 @@
+        print("E) Walk East")
+        print("W) Walk West")
+        print("GH) Go Home")
+
```
Numbering these options might be more intuitive so the user can just leave their hand on the number row or numpad.

Gizmofire committed:
Added peer review summary

Gizmofire committed:
Added excel sheet

JuanEstrada1 commented on game.py:
```diff
@@ -0,0 +1,271 @@
+def get_position_Radius():
+    return math.sqrt(position[0]**2 + position[1]**2)
+
+# Searches for loot from fallen enemies and/or ground
```
Lines 90 - 115: Lines of codes could possibly be condensed and refactored as to make things cleaner and simpler.

    def search_loot(bp):
        angle = get_Angle()  # Kept but unused
        distance = get_position_Radius()  # Kept but unused
        loot_number = random.randint(1, 5)
        amount = random.randint(1, 3)
    
        if loot_number == 1:
            print(f"You found {amount} stick{'s' if amount > 1 else ''}")
            bp.add_item("stick", amount)
        elif loot_number == 2:
            print(f"You found {amount} rock{'s' if amount > 1 else ''}")
            bp.add_item("rock", amount)
        else:
            print("You found nothing")

JuanEstrada1 commented on game.py:
```diff
@@ -0,0 +1,271 @@
+        print("5) Exit and Save")
+        print("10) HARD RESET BACKPACK")
+        
+        choice = input("Select an option (1-5): ")
```
Lines 130 - 155: Possible dictionary can be arranged so that the choices can be organized clearer without repetitive use of the "elif" command.

    choice = input("Select an option (1-5, 10): ")

    options = {
        "1": (go_Home, "[Exploring the world...]"),
        "2": (bp.to_table, "[Opening your backpack...]"),
        "3": (view_item_description, "[Viewing item description...]"),
        "4": (crafting_menu, "[Crafting items...]"),
        "5": (bp.save_backpack, "[Game saved. Exiting...]"),
        "10": (bp.reset_backpack, "[Resetting Backpack]")
    }
    
    if choice in options:
        action, message = options[choice]
        print(f"\n{message}\n")
        action()
        if choice == "5":
            play = False 
    else:
        print("\nInvalid choice. Please select a number between 1 and 5 or 10.")

Gizmofire committed:
Added comment to location of peer review

Gizmofire committed:
updated peer review code

ThomasHakwins committed:
Add files via upload

Gizmofire committed:
Fixed all the comments except new function considereations for game.py

Gizmofire committed:
Merge branch 'main' of https://github.com/ETE4990-S25/project-1-gordon-and-thomas-in-the-mornin

Gizmofire commented:
Waiting for comments from 3 people: edgar2899, KTauroi, and joeyenriquez.

KTauroi commented on items_and_backpack/backpack.py:
```diff
@@ -0,0 +1,123 @@
+                else:
+                    raise ValueError("Not enough items to remove")
+                return
+        raise ValueError("Item not found in backpack")
```
I think you could set it so that count passed is greater than or equal to count present, the count present becomes 0 as a nested if statement rather than these 2.

KTauroi commented on items_and_backpack/backpack.py:
```diff
@@ -0,0 +1,123 @@
+                if choice.lower() == "y":
+                    self.capacity = 7
+                    self.remove_item("rock", 5)
+                    print("Backpack upgraded to a capacity of 7")
```
The logic for upgrading capacity is very similar could probably be refactored so that you're not repeating yourself, maybe store the upgrade variables in something like a list of dictionaries.

Also keep consistent capitalization of Backpack

KTauroi commented on items_and_backpack/backpack.py:
```diff
@@ -0,0 +1,123 @@
+        else:
+            print("You already have the max backpack size!!!!")
+
+    def save_backpack(self):
```
Have some way to handle an error if there's a writing error with json.dump

KTauroi commented on items_and_backpack/backpack.py:
```diff
@@ -0,0 +1,123 @@
+    def to_table(self):
+        print("Item \t\t| Count \n----------------")
+        for item in self.items:
+            if item is None:
```
is none seems unnecessary if you're already passing something to this function

KTauroi commented on items_and_backpack/items.py:
```diff
@@ -0,0 +1,61 @@
+    
+    def get_item_rarity(self):
+        return self.rarity
+    
```
Not sure if I'm missing it but this class isn't used anywhere else in the code. So it's probably fine to delete this Item class

ThomasHakwins committed:
fixed line 27

ThomasHakwins committed:
refactored code added

Commit:
refactored code for crafting class

Commit:
backpack class fixed and refactored

Commit:
fixed backpack with edge case and tests
